src/main.c

```diff
@@ -6,54 +6,37 @@
 #include "algo_ui.h"
 #include "algo_addr.h"
 #include "algo_tx.h"
+#include "command_handler.h"
+#include "apdu_protocol_defines.h"
 
 unsigned char G_io_seproxyhal_spi_buffer[IO_SEPROXYHAL_BUFFER_SIZE_B];
 
-#define OFFSET_CLA    0
-#define OFFSET_INS    1
-#define OFFSET_P1     2
-#define OFFSET_P2     3
-#define OFFSET_LC     4
-#define OFFSET_CDATA  5
 
-#define CLA           0x80
 
-#define P1_FIRST 0x00
-#define P1_MORE  0x80
-#define P1_WITH_ACCOUNT_ID  0x01
-#define P1_WITH_REQUEST_USER_APPROVAL  0x80
-
-#define P2_LAST  0x00
-#define P2_MORE  0x80
-
-#define INS_SIGN_PAYMENT    0x01    // Deprecated, unused
-#define INS_SIGN_KEYREG     0x02    // Deprecated, unused
-#define INS_GET_PUBLIC_KEY  0x03
-#define INS_SIGN_PAYMENT_V2 0x04
-#define INS_SIGN_KEYREG_V2  0x05
-#define INS_SIGN_PAYMENT_V3 0x06
-#define INS_SIGN_KEYREG_V3  0x07
-#define INS_SIGN_MSGPACK    0x08
 
 /* The transaction that we might ask the user to approve. */
 txn_t current_txn;
-already_computed_key_t current_pubkey;
+
 
 /* A buffer for collecting msgpack-encoded transaction via APDUs,
  * as well as for msgpack-encoding transaction prior to signing.
  */
+
 #if defined(TARGET_NANOX)
-static uint8_t msgpack_buf[2048];
+#define TNX_BUFFER_SIZE 2048
+static uint8_t msgpack_buf[TNX_BUFFER_SIZE];
 #else
-static uint8_t msgpack_buf[900];
+#define TNX_BUFFER_SIZE 900
+static uint8_t msgpack_buf[TNX_BUFFER_SIZE];
```

Comment:
lines 27 and 30 are duplicative, consider taking them out of the if to avoid that.